File: src/main/java/org/academy/api/client/gui/command/ImageCircleDrawCommand.java
package org.academy.api.client.gui.command;

import com.mojang.blaze3d.textures.GpuSampler;
import com.mojang.blaze3d.textures.GpuTextureView;
import org.academy.api.client.Render;
import org.academy.api.client.render.TextureBinding;

import java.util.List;

public class ImageCircleDrawCommand extends PosTexColorRectDrawCommand {
    protected final GpuTextureView texture;
    protected final GpuSampler sampler;

    public ImageCircleDrawCommand(
            GpuTextureView texture,
            GpuSampler sampler,
            float width,
            float height,
            float u0,
            float v0,
            float u1,
            float v1,
            float red,
            float green,
            float blue,
            float alpha
    ) {
        super(Render.RenderPipelines.IMAGE_CIRCLE,
                width, height,
                u0, v0, u1, v1,
                red, green, blue, alpha,
                List.of(new TextureBinding("Sampler0", texture, sampler)),
                List.of()
        );
        this.texture = texture;
        this.sampler = sampler;
    }
}

File: src/main/kotlin/org/academy/api/client/gui/command/ImageCircleDrawCommand.kt
package org.academy.api.client.gui.command

import com.mojang.blaze3d.textures.GpuSampler
import com.mojang.blaze3d.textures.GpuTextureView
import org.academy.api.client.render.Render
import org.academy.api.client.render.TextureBinding

class ImageCircleDrawCommand : PosTexColorRectDrawCommand {
    constructor(
        texture: GpuTextureView,
        sampler: GpuSampler,
        width: Float,
        height: Float,
        u0: Float,
        v0: Float,
        u1: Float,
        v1: Float,
        u2: Float,
        v2: Float,
        u3: Float,
        v3: Float,
        red: Float,
        green: Float,
        blue: Float,
        alpha: Float
    ) : super(
        Render.RenderPipelines.IMAGE_CIRCLE,
        width, height,
        u0, v0, u1, v1, u2, v2, u3, v3,
        red, green, blue, alpha,
        listOf(
            TextureBinding(
                "Sampler0",
                texture,
                sampler
            )
        ),
        mutableListOf()
    )
}
